aspnetcore/fundamentals/middleware/request-decompression.md

```diff
@@ -0,0 +1,83 @@
+
+If the middleware encounters a request with compressed content but is unable to decompress it, the request is passed to the next delegate in the pipeline. For example, a request with an unsupported `Content-Encoding` header value or multiple `Content-Encoding` header values, is passed to the next delegate in the pipeline.
+
+## Configuration
+
+The following code shows how to enable request decompression for the [default](#default) `Content-Encoding` types:
+
+[!code-csharp[](samples/request-decompression/7.x/Program.cs?name=snippet_WithDefaultProviders&highlight=3,7)]
+
+<a name="default"></a>
+
+## Default decompression providers
+
+The `Content-Encoding` header values that the request decompression middleware supports by default are listed in the following table:
+
+| [`Content-Encoding`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding) header values | Description |
+| --------- | --------- |
+| `br`      | [Brotli compressed data format](https://tools.ietf.org/html/rfc7932) |
+| `deflate` | [DEFLATE compressed data format](https://tools.ietf.org/html/rfc1951) |
+| `gzip`    | [Gzip file format](https://tools.ietf.org/html/rfc1952) |
+
+## Custom decompression providers
+
+Support for custom encodings can be added by creating custom decompression provider classes that implement <xref:Microsoft.AspNetCore.RequestDecompression.IDecompressionProvider>:
+
+[!code-csharp[](samples/request-decompression/7.x/CustomDecompressionProvider.cs?name=snippet_CustomDecompressionProvider)]
+
+Custom decompression providers are registered with <xref:Microsoft.AspNetCore.RequestDecompression.RequestDecompressionOptions> along with their corresponding `Content-Encoding` header values:
+
+[!code-csharp[](samples/request-decompression/7.x/Program.cs?name=snippet_WithCustomProvider&highlight=3-6,10)]
+
+## Request size limits
+
+In order to guard against [zip bombs or decompression bombs](https://en.wikipedia.org/wiki/Zip_bomb):
+
+* The maximum size of the decompressed request body is limited to the request body size limit enforced by the endpoint or server.
+* If the number of bytes read from the decompressed request body stream exceeds the limit, an [InvalidOperationException](xref:System.InvalidOperationException) is thrown to prevent additional bytes from being read from the stream.
+
+The maximum request size for an endpoint is set by:
+
+* <xref:Microsoft.AspNetCore.Http.Metadata.IRequestSizeLimitMetadata.MaxRequestBodySize?displayProperty=nameWithType>, such as <xref:Microsoft.AspNetCore.Mvc.RequestSizeLimitAttribute> or <xref:Microsoft.AspNetCore.Mvc.DisableRequestSizeLimitAttribute> for MVC endpoints.
+* If the preceding API's aren't applied, the global server size limit specified by <xref:Microsoft.AspNetCore.Http.Features.IHttpMaxRequestBodySizeFeature.MaxRequestBodySize?displayProperty=nameWithType> is used.
```

Comment:
The default size limit is set at the server level such as 
https://github.com/dotnet/aspnetcore/blob/197c1693d3c830af52b587e8d88891bc9689be44/src/Servers/Kestrel/Core/src/KestrelServerLimits.cs#L148-L157